pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/documentation/AbstractCommentRule.java

```diff
@@ -202,22 +202,26 @@ private boolean isCommentBefore(FormalComment n1, Node n2) {
 
         SortedMap<Integer, Node> itemsByLineNumber = new TreeMap<>();
 
-        List<ASTClassOrInterfaceDeclaration> packageDecl = cUnit
-                .findDescendantsOfType(ASTClassOrInterfaceDeclaration.class);
-        addDeclarations(itemsByLineNumber, packageDecl);
+        List<ASTClassOrInterfaceDeclaration> classDecl = new ArrayList<>();
```

Comment:
I still think it would be nice to remove that extra statement by adding an overload `<T> List<T> findDescendantsOfType(Class<T>, boolean)`. This would be consistent with the signature and usage of all other `find*` methods -- which return a list instead of `void`. These variables could then be initialized with a single expression

Reply:
This has been done on `AbstractNode`, but adding it to the interface will have to wait to 7.0.0 (add it to #881 when merging). The same applies to `PreserveStackTraceRule` where such a change could not be taken advantage of due to the usage of `jjtGetParent` which returns `Node`